src/System.Private.ServiceModel/src/System/ServiceModel/Channels/HttpChannelFactory.cs

```diff
@@ -904,6 +904,9 @@ public HttpClientChannelAsyncRequest(HttpClientRequestChannel channel)
                     _httpClient = await _channel.GetHttpClientAsync(_to, _via, _timeoutHelper);
                     _httpRequestMessage = _channel.GetHttpRequestMessage(_via);
 
+                    // The _httpRequestMessage field will be set to null by Cleanup() due to faulting
+                    // or aborting, so use a local copy for exception handling within this method.
+                    HttpRequestMessage httpRequestMessage = _httpRequestMessage;
```

Comment:
NIT: I'd rather see the local var being set first, and then the field in the class set after, e.g.: 

```csharp

HttpRequestMessage httpRequestMessage = _channel.GethttpRequestMessage(_via); 
_httpRequestMessage = httpRequestMessage

```

More of a stylistic thing given the timings here, but ensures that we always manage to capture the httpRequestMessage before anyone else has a chance to muck with it.